### PR TITLE
Support design icons

### PR DESCRIPTION
**Pull request by chogorma**
## Description

Support design icons

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/icon/Icon.ts [hunk 1]
```diff
@@ -40,6 +42,11 @@ export namespace Icon {
     @property({ type: String }) ariaHidden: any;
     @property({ type: Boolean }) isActive = false;
     @property({ type: Boolean }) isComboBoxIcon = false;
+    @property({ type: Boolean }) enabled = false; // enable design icon lookup
```
Comment (chogorma) [inline web-components/src/components/icon/Icon.ts:45]: We will at some point have this always enabled.

File: web-components/src/components/icon/Icon.ts [hunk 2]
```diff
@@ -40,6 +42,11 @@ export namespace Icon {
...
+    @property({ type: Boolean }) override = false; // use icon as design icon
```
Comment (chogorma) [inline web-components/src/components/icon/Icon.ts:46]: `override` is to allow us to just the the design icons without doing a lookup.

**Review by NotNestor — APPROVED** [review]

File: web-components/src/components/icon/Icon.stories.ts [hunk 3]
```diff
@@ -60,9 +61,11 @@ export const Icon = () => {
         .type=${type}
         .size=${size}
         .sizeOverrided=${sizeOverrided}
-        @icon-click=${(action('dispatchEvent'))}>
+        .override=${override}
+        .designEnabled=${enable}
+        @icon-click=${action("dispatchEvent")}
```
Comment (NotNestor) [inline web-components/src/components/icon/Icon.stories.ts:66]: incorrect syntax should be ?designEnabled

File: web-components/src/components/icon/Icon.stories.ts [hunk 4]
```diff
@@ -39,7 +39,8 @@ export default {
 };
 
 export const Icon = () => {
-  const darkTheme = boolean("darkMode", false);
+  const enable = boolean("Design Enabled", false);
+  const darkTheme = boolean("Dark Mode", false);
   const lumos = boolean("Lumos Theme", false);
   const theme = select("Theme name", ThemeNameValues, "");
```
Comment (NotNestor) [inline web-components/src/components/icon/Icon.stories.ts:45]: can you remove lumos from here and set theme to a default of lumos

**Review by NotNestor — APPROVED** [review]

**Pull request merged**